agent/utils/ephemeral_ports.go

```diff
@@ -182,3 +161,38 @@ func getHostPortRange(numberOfPorts, start, end int, protocol string) (string, i
 
 	return fmt.Sprintf("%d-%d", resultStartPort, resultEndPort), resultEndPort, nil
 }
+
+var isPortAvailableFunc = isPortAvailable
```

Comment:
are we setting this var only for testing `getHostPortRange`?  Are we trying to circumvent the net.ListenPacket/Listen calls?  I'd like to try using gomock somehow to do this override for testing rather than baking in a function override.